### DISCUSSION TITLE
blog/draw-a-word-cloud-with-a-shiny-app/

### DISCUSSION THREAD

**Discussion by giscus** [AntoineSoetewey/statsandr#89] [General]
# blog/draw-a-word-cloud-with-a-shiny-app/

Word clouds are often used in a text mining analysis. See how to draw a word cloud in R thanks to a Shiny app. See also some examples or use your own dataset

https://statsandr.com/blog/draw-a-word-cloud-with-a-shiny-app/

**Comment by lomper1212** [comment]
For a very non-savvy user: what modifications would be needed so that phrases are kept instead of single words? Say with a cvs file?

**Reply by lomper1212** [reply]
And also, I have a corpus of some 100 words. The main ones do not appear all the time, each time I refresh there's a different one that is left out.

The program is gorgeous though

**Reply by AntoineSoetewey** [reply]
Hello,

> For a very non-savvy user: what modifications would be needed so that phrases are kept instead of single words? Say with a cvs file?

You would need to edit the `create_wordcloud()` function (see code [here](https://github.com/AntoineSoetewey/word-cloud/blob/71a6d1d8146439ba34598132eee6b373d52cb409/app.R#L155)). Feel free to propose a merge request if you happen to edit it.

> And also, I have a corpus of some 100 words. The main ones do not appear all the time, each time I refresh there's a different one that is left out.

This [code](https://github.com/AntoineSoetewey/word-cloud/blob/71a6d1d8146439ba34598132eee6b373d52cb409/app.R#L184-L187) takes the _n_ most common words (i.e., the _n_ most frequent words), so if you see it once, you should see it every time. It is true that they do not appear at the same place and of the same color, but they should always be the same (try with 3 words for instance, you will see that they are always the 3 same). Can you make sure they do not appear somewhere else?

> The program is gorgeous though

Thanks !

**Comment by lomper1212** [comment]
Thanks a lot Antoine. I'll let you know if I get to modify the code. I tried again but there is something not working. In the text sample I provide, the word agroecology never appears in the wordcloud, whereas transitions and territorial tend to disappear at times. Thanks!

https://drive.google.com/file/d/1Y-OibTk_uTih5bT1vwxxFqXK-yPuQXb3/view?usp=sharing

**Reply by AntoineSoetewey** [reply]
I tried myself, you're right the word agroecology does not appear. I thus replaced agroecology by ecology, and then agroecology by agro. In both cases it worked.

So it must be something with the way the `create_wordcloud()` function handles words that are not recognized.

Have you tried with another wordcloud generator? Do you have the same issue with this word?

Regards,
Antoine

**Reply by lomper1212** [reply]
I tried other non-R ones, but the result is much less attractive than this one... thanks a lot!
